client/src/main/java/io/pravega/client/control/impl/Controller.java

```diff
@@ -118,6 +118,46 @@
      */
     CompletableFuture<Boolean> updateStream(final String scope, final String streamName, final StreamConfiguration streamConfig);
 
+    /**
+     * API to add a Subscriber to the Stream.
+     * @param scope Scope name
+     * @param streamName Stream name
+     * @param subscriber Name/Id that uniquely identifies a Stream Subscriber.
+     * @return A future which will throw if the operation fails, otherwise returning a boolean to
+     *         indicate that the subscriber was updated in Stream Metadata.
+     */
+    CompletableFuture<Boolean> addSubscriber(final String scope, final String streamName, final String subscriber);
+
+    /**
+     * API to remove a Subscriber from list of Subscribers for the Stream.
+     * @param scope Scope name
+     * @param streamName Stream name
+     * @param subscriber Name/Id that uniquely identifies a Stream Subscriber..
+     * @return A future which will throw if the operation fails, otherwise returning a boolean to
+     *         indicate that the subscriber was updated in Stream Metadata.
+     */
+    CompletableFuture<Boolean> removeSubscriber(final String scope, final String streamName, final String subscriber);
+
+    /**
+     * Get list of Subscribers for the Stream.
+     * @param scope Scope name
+     * @param streamName Stream name
+     * @return List of StreamSubscribers
+     */
+    CompletableFuture<List<String>> getSubscribersForStream(final String scope, final String streamName);
+
+    /**
+     * API to update the truncation StreamCut for a particular Subscriber on Controller.
+     * Used when Stream has Consumption Based Retention Policy configured.
+     * @param scope Scope name
+     * @param streamName Stream name
+     * @param subscriber Name/Id that uniquely identifies a Stream Subscriber..
+     * @param streamCut StreamCut at which Stream can be Truncated for a Consumption based retention policy
+     * @return A future which will throw if the operation fails, otherwise returning a boolean to
+     *         indicate that the subscriber was updated in Stream Metadata.
+     */
+    CompletableFuture<Boolean> updateTruncationStreamCut(final String scope, final String streamName, final String subscriber, final StreamCut streamCut);
```

Comment:
nit: since this api is in the context of a subscriber, the name should reflect it.. perhaps a better name for this API would be `reportSubscriberStreamCut` or `reportSubscriberPosition`.

Reply:
Ok will change.